docs/ROADMAP.md

```diff
@@ -0,0 +1,47 @@
+# **Intro**
+
+The last year has been an eventful one for Anvil, with much of the focus on fixing incremental compilation issues and fortifying our integration test capabilities. There has also been a lot of discussion and investigations around the different paths forward in terms of both K2 support and the long-term vision of the library. This doc aims to provide some transparency around where things have settled, where we’re headed, and why various choices were made.
+
+# **Roadmap**
+
+The long term vision is currently bucketed into three major milestones:
+
+1. Introduce support for FIR and the K2 compiler. This will tentatively happen in two phases:  
+   1. K2 baseline support for all primary Anvil functionality (contributions and merging), with KAPT utilized instead of KSP. (For those that need Dagger-KSP interop or K2 support now, we endorse Zac Sweer’s Anvil-KSP fork: [https://github.com/ZacSweers/anvil](https://github.com/ZacSweers/anvil).)  
+   2. K2 support for generating dagger factories.  
+2. Anvil 3.0 \- Introduce Anvil Component generation, remove dependency on both Dagger and KAPT. At this point Anvil will become a completely independent Kotlin compiler plugin which broadly follows the existing contract and API surface. We don't currently have plans for KMP support since it's not something we need internally, but it's something we're open to re-evaluating after the release of Anvil 3.0.  
+3. Long-tail: Evolve the API to match new capabilities and take advantage of our lack of dependency on Dagger, with a focus on ease of migration for consumers.
+
+# **Context**
+
+This section focuses on providing context for how we ended up with the roadmap items above and the decisions made along the way.
+
+## **Breaking from Dagger**
+
+The decision to break our dependency on Dagger stems from a number of constraints Anvil is limited by today, including:
```

Comment:
Error reporting, too, right? Library users will always get Dagger errors, which they then have to interpret alongside what might be happening in Anvil

Reply:
Good catch! Added a point for error reporting